SimplenoteUITests/SimplenoteUITestsHistory.swift

```diff
@@ -0,0 +1,231 @@
+
+        trackStep()
+        History.close()
+        NoteEditor.leaveEditor()
+        NoteList.trashNote(noteName: initialNoteText)
+    }
+
+    func testRestoredVersionIsAddedOnTopOfTheOtherChanges() throws {
+        trackTest()
+        let initialNoteText = notePrefix + getRandonId()
+        let editedNoteText = notePrefix + getRandonId()
+
+        trackStep()
+        NoteList.addNoteTap()
+        NoteEditorAssert.editorShown()
+
+        trackStep()
+        NoteEditor.clearAndEnterText(enteredValue: initialNoteText)
+        NoteEditor.waitBeforeEditingText(delayInSeconds: 3, newText: editedNoteText)
+
+        trackStep()
+        NoteEditor.openHistory()
+        HistoryAssert.historyShown()
+        History.setSliderPosition(position: 0.0)
+        History.restoreNote()
+        NoteEditorAssert.textViewWithExactValueShownOnce(value: initialNoteText)
+
+        trackStep()
+        NoteEditor.openHistory()
+        HistoryAssert.historyShown()
+        History.setSliderPosition(position: 1.0)
+        NoteEditorAssert.textViewWithExactValueShownOnce(value: initialNoteText)
+
+        trackStep()
+        History.setSliderPosition(position: 0.5)
+        NoteEditorAssert.textViewWithExactValueShownOnce(value: editedNoteText)
+
+        trackStep()
+        History.setSliderPosition(position: 0.0)
+        NoteEditorAssert.textViewWithExactValueShownOnce(value: initialNoteText)
+
+        trackStep()
+        History.close()
+        NoteEditor.leaveEditor()
+        NoteList.trashNote(noteName: initialNoteText)
+    }
+
+    func testPreviousNoteIsNotRestoredWhenTheHistoryPanelIsDismissed() throws {
+        trackTest()
+        let initialNoteText = notePrefix + getRandonId()
+        let editedNoteText = notePrefix + getRandonId()
+
+        trackStep()
+        NoteList.addNoteTap()
+        NoteEditorAssert.editorShown()
+
+        trackStep()
+        NoteEditor.clearAndEnterText(enteredValue: initialNoteText)
+        NoteEditor.waitBeforeEditingText(delayInSeconds: 3, newText: editedNoteText)
+
+
+        trackStep()
+        NoteEditor.openHistory()
+        HistoryAssert.historyShown()
+
+        trackStep()
+        History.setSliderPosition(position: 0.0)
+        History.close()
+        NoteEditorAssert.textViewWithExactValueShownOnce(value: editedNoteText)
+
+        trackStep()
+        NoteEditor.leaveEditor()
+        NoteList.trashNote(noteName: editedNoteText)
+    }
+
+    func testHistoryIsKeptAfterRecoveringNoteFromTrash() throws {
+        trackTest()
+        let initialNoteText = notePrefix + getRandonId()
+        let editedNoteText = notePrefix + getRandonId()
+
+        trackStep()
+        NoteList.addNoteTap()
+        NoteEditorAssert.editorShown()
+
+        trackStep()
+        NoteEditor.clearAndEnterText(enteredValue: initialNoteText)
+        NoteEditor.waitBeforeEditingText(delayInSeconds: 3, newText: editedNoteText)
+
+        trackStep()
+        getToAllNotes()
```

Comment:
`getToAllNotes()` is used in setUp/tearDown methods, since we don't always know where the app might be. It will firstly try to close the WebView which might be shown in note preview mode, then leave the preview, and then leave the editor.

However, in this test, we do know that the way to get to all note is this one:

```suggestion
        NoteEditor.leaveEditor()
```

And this will save us some time by not trying to tap controls that we know do not exist anyway.